### ISSUE TITLE
L-BFGS iteration records and metrics

### ISSUE THREAD

**Issue by AmirNoori68** [lululxvi/deepxde#1605]
Hi,
I have two questions.
First:
An option named "display_every" for Adam optimizer exists that records the results as we want.
But how about L-BFGS? It records the results every 1000 iterations. How do we change it to something else, such as 100 iterations?
I tried 
"
model.compile("L-BFGS")
model.train(display_every=100)
"
and it is not working.

Second:
How do we get "metrics=["l2 relative error"]" when using L-BFGS? 
Such a thing 
"
model.compile("L-BFGS"metrics=["l2 relative error"])
"
is not working.

Leading to an error like:
"
  File ~\anaconda3\envs\deepxde2\lib\site-packages\deepxde\metrics.py:12 in l2_relative_error
    return np.linalg.norm(y_true - y_pred) / np.linalg.norm(y_true)

TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
"

as I tried for the heat or burger's equations (forward problems).
and
what else metric we can get and what to put here:
"
model.compile("L-BFGS",metrics=["?????"])
"

Thanks for your time.

**Comment by tsarikahin** [comment]
Yes, I believe the Pytorch implementation has a bug that it does not improve loss accuracy using LBFGS so switch to tf. You will get NaN at the end as well.

**Comment by lululxvi** [comment]
Yes, this bug was introduced for PyTorch 2.x. If you use PyTorch 1.x or TensorFlow, it should be fine.
